### ISSUE TITLE
Error: Download failed: Authorisation is required to install the packages.

### ISSUE THREAD

**Issue by mo3rfan** [mas-cli/mas#397]
On m1 Big Sur, mas version 1.8.2, I get the following when I try to update an app.

```
Upgrading 1 outdated application:
Developer (9.0.1)
==> Downloading Apple Developer
Error: Download failed: Authorisation is required to install the packages.
```

I am signed in via App store and I have the app 'purchased' from the app store (the app is moved to a different drive though).

I have also tried `mas reset`

**Comment by MikeiLL** [comment]
Same here with Xcode. I'm trying to upgrade from `Xcode (11.3.1)`. Had been on Mojave, previously.

```
Upgrading 1 outdated application:
Xcode (11.3.1)
==> Downloading Xcode
```

The annoying thing is, it doesn't provide the error message until after it has downloaded the entire thing. Tried running with `sudo`, which returned

```
Warning: Nothing found to upgrade
```

I'm excited to get into this tool. Just discovered it.

**Comment by MikeiLL** [comment]
Wait a minute! After a more thorough look at the README, `mas upgrade` will upgrade all of the available App Store apps, but to upgrade a single app, it wants the "App Identifier", which is listed in the output of `mas outdated`

For me:

```
634148309 Logic Pro X (10.5.1 -> 10.6.2)
1295203466 Microsoft Remote Desktop (10.6.6 -> 10.6.7)
409201541 Pages (10.0 -> 11.1)
497799835 Xcode (11.3.1 -> 12.5.1)
```

Interestingly, running `mas upgrade Pages` seems to have worked:

```
Upgrading 1 outdated application:
Pages (10.0)
==> Downloading Pages
==> Installed Pages
$ mas list
# abridged...
409201541   Pages                        (11.1)
```

So now I'm running `mas upgrade 497799835`. Will know in 30 minutes, or maybe tomorrow morning, if it worked or not.

**Comment by MikeiLL** [comment]
Same error.

**Comment by chris-araman** [comment]
@MikeiLL:
`sudo` shouldn't be necessary, and may actually be problematic.


1. Are you signed into the App Store app (the GUI)?
1. Are you signed in with the same Apple ID as originally purchased and installed Xcode, Pages, or any other apps that report this error?
1. Might the app(s) have been installed by another local Mac user account?
1. Are the apps being upgrade installed to the `/Applications/` path, or are there other copies found elsewhere on your system, perhaps in a Time Machine backup?
1. You mentioned being on Mojave previously. What version of macOS are you currently running?

**Comment by MikeiLL** [comment]
Thank you, Chris!

> 
> 1. Are you signed into the App Store app (the GUI)?
    No, then yes.
> 2. Are you signed in with the same Apple ID as originally purchased and installed Xcode, Pages, or any other apps that report this error?
    Yes. I only have one, as far as I know.
> 3. Might the app(s) have been installed by another local Mac user account?
    I don't believe so.
> 4. Are the apps being upgrade installed to the `/Applications/` path, or are there other copies found elsewhere on your system, perhaps in a Time Machine backup?
    Yes! I have Xcode on an external drive. Is that something `mas` can currently support?
> 5. You mentioned being on Mojave previously. What version of macOS are you currently running?
    Big Sur. (So far so good)

Thank you very much, man.

I have installed Xcode through the GUI, and moved it to the external drive. Mas-cli still lists it. Well done! Any suggestions for next time I need to update?

**Comment by chris-araman** [comment]
Ah, I bet that's what's causing the issue. See #309 for more details. After https://github.com/mas-cli/mas/commit/d39058c7be405944fa1a172f39e5168fb9cef866, not yet released, we'll no longer report apps outside `/Applications/` as outdated, or try to upgrade them.

I'm not sure what would cause this particular error exactly, but it does seem like the App Store really prefers that everything is in `/Applications/`. `mas` is built against the same PrivateFrameworks used by the App Store, so it has similar limitations. That said, we're open to pull requests if you find a way to safely work around those limitations. See [`CONTRIBUTING.md`](https://github.com/mas-cli/mas/blob/main/CONTRIBUTING.md) for details.

**Comment by MikeiLL** [comment]
Fantastic. It's too bad Apple isn't supporting management of nearly 20Gig apps on an external drive. Perhaps they, and by extension mas-cli will in the future. In the meantime, I'll have to just use it for apps located in Applications. And will look forward to your upgrade. Hopefully I'll be able to contribute more significantly in the future. Great work, sir.

**Comment by chris-araman** [comment]
Closing, as both reports seem to be related to #309 and should be addressed by d39058c.

**Comment by cognivator** [comment]
A few years later, similar epiphany: sign into the App Store GUI first (this is a clean reinstall). Being signed in only to iCloud during the fresh setup wizard is apparently not enough. 🙂
Monterey 12.7
